### ISSUE TITLE
glib upgrade 2.64.2_1 to 2.64.3 on Catalina can't install - deprecated method

### ISSUE THREAD

**Issue by blogabe** [Homebrew/homebrew-core#57595]
# Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

`brew upgrade glib`
Tried installing from source as well as force bottle and neither worked

## What happened (include command output)

`brew upgrade glib`

<details>
  <summary>Command output</summary>
  <pre>

% brew upgrade glib
==> Upgrading 1 outdated package:
glib 2.64.2_1 -> 2.64.3
==> Upgrading glib 2.64.2_1 -> 2.64.3
==> Downloading https://raw.githubusercontent.com/Homebrew/formula-patches/61642
Already downloaded: /Users/me/Library/Caches/Homebrew/downloads/54f3f612bdfc9bfc09633176c7673816fe2a2188e470bf48d60d826bdf6bf21a--hardcoded-paths.diff
==> Downloading https://download.gnome.org/sources/glib/2.64/glib-2.64.3.tar.xz
Already downloaded: /Users/me/Library/Caches/Homebrew/downloads/1aa18c61373e14cf15da7bd78ea89324c8c9f379b45060f48c4dc4317c8a4e77--glib-2.64.3.tar.xz
==> Patching
==> Applying hardcoded-paths.diff
patching file gio/gdbusprivate.c
Hunk #1 succeeded at 2471 (offset 364 lines).
Hunk #2 succeeded at 2481 (offset 364 lines).
patching file gio/xdgmime/xdgmime.c
patching file glib/gutils.c
Hunk #1 succeeded at 2497 (offset 417 lines).
Error: An exception occurred within a child process:
  MethodDeprecatedError: Calling Language::Python.rewrite_python_shebang is deprecated! Use Utils::Shebang.rewrite_shebang and Shebang.python_shebang_rewrite_info(python_path) instead.
Please report this issue to the homebrew/core tap (not Homebrew/brew or Homebrew/core), or even better, submit a PR to fix it:
  /Users/me/src/homebrew/xpfx/Library/Taps/homebrew/homebrew-core/Formula/glib.rb:35

  </pre>
</details>

## What you expected to happen

upgrade

## Step-by-step reproduction instructions (by running `brew install` commands)

`brew install glib`

brew config is
```
% brew config
HOMEBREW_VERSION: 2.4.3-99-g9cce4c5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9cce4c5be3a740dfa6b3d14fc14eadec29b1657c
Last commit: 10 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3c4b961e59a36ee0fc496ab8f890108292c4129b
Core tap last commit: 25 minutes ago
HOMEBREW_PREFIX: /Users/me/src/homebrew/xpfx
HOMEBREW_REPOSITORY: /Users/me/src/homebrew/xpfx
HOMEBREW_CELLAR: /Users/me/src/homebrew/xpfx/Cellar
HOMEBREW_DEVELOPER: set
HOMEBREW_EDITOR: /Users/me/local/bin/mate -w
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_NO_ANALYTICS: set
CPU: 16-core 64-bit kabylake
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
Clang: 11.0 build 1103
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.5-x86_64
CLT: 11.5.0.0.1.1588476445
Xcode: 11.5
```

brew doctor is:
```
% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/me/homebrew/bin/pkg-config
  /Users/me/homebrew/bin/freetype-config
  /Users/me/homebrew/bin/libpng16-config
  /Users/me/homebrew/bin/libpng-config

Warning: Your Homebrew's prefix is not /usr/local.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/usr/local).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience while you are running this unsupported configuration.
```

**Comment by Rylan12** [comment]
Can you use the 2.64.3 bottle instead of building from source?

This is being fixed for version 2.64.4 (in progress in #57584). I believe the PR is blocked, though, so it might be a little bit before it is merged (I guess if it's really needed, a revision can be made to 2.64.3 in the meantime).

**Comment by SMillerDev** [comment]
Please see the end of brew doctor:

> Warning: Your Homebrew's prefix is not /usr/local.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/usr/local).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience while you are running this unsupported configuration.

**Comment by blogabe** [comment]
> Can you use the 2.64.3 bottle instead of building from source?
> 
> This is being fixed for version 2.64.4 (in progress in #57584). I believe the PR is blocked, though, so it might be a little bit before it is merged (I guess if it's really needed, a revision can be made to 2.64.3 in the meantime).

Yep... no worries. Just passing along the info.

**Comment by blogabe** [comment]
> Please see the end of brew doctor:
> 
> > Warning: Your Homebrew's prefix is not /usr/local.
> > Some of Homebrew's bottles (binary packages) can only be used with the default
> > prefix (/usr/local).
> > You will encounter build failures with some formulae.
> > Please create pull requests instead of asking for help on Homebrew's GitHub,
> > Discourse, Twitter or IRC. You are responsible for resolving any issues you
> > experience while you are running this unsupported configuration.

I acknowledge you're a long time maintainer that has contributed much to the project and you deal with a high volume of support issues. I believe your answer is a cop out. Sure, you are under no obligation to provide help. But the issue isn't because Homebrew is installed in a folder outside of `/usr/local`.

I also have been using Homebrew since Max was the maintainer in non default locations. The formula clearly installed before. And there's an expectation that it should upgrade. What's the suggestion here... to recreate it the issue in the default location? Would that have changed anything. Your reply - while within bounds - doesn't really address the error.

Maintainers have continually requested users act with the right tone and won't respond to hostile comments. It's disappointing that this is one sided and I say that based on my direct and indirect experience.

**Comment by SMillerDev** [comment]
> I believe your answer is a cop out. Sure, you are under no obligation to provide help. 
With the current maintenance burden of core, sometimes those are required to help the majority of users.

> But the issue isn't because Homebrew is installed in a folder outside of /usr/local.
I don't know that based on this issue, and there have been so many issues with source builds that it's impossible to keep track of all of them. This is why the doctor message is there (along with other messages in other places).

> What's the suggestion here... to recreate it the issue in the default location?
Yes, that would make sure that it's indeed an issue in the source and not in the device configuration.

> It's disappointing that this is one sided
While my communication style is far from perfect, and I know that I can be better in this regard. I'd like to point out that people ignoring of the points in the issue template makes it a lot harder to maintain the project. If everyone would stick with the rules set out by the project I'm sure everything would be easier for everyone.

**Comment by blogabe** [comment]
Which points of the issue template were ignored? I ran update and doctor and while I didn't provide gist-logs, I did provide the details for config and doctor. Is there anything additional in the template that's needed?

The error message clearly states there's a deprecated method so still not sure how homebrew being installed in a custom directory is related. Furthermore, I was doing what was asked of me in the error message.... like if you're just going to be so dismissive then there's no point in actually providing the requested feedback.

No need to answer the above questions... I've spent more time on this that I care to.
